show/fgnhg.py

```diff
@@ -47,123 +41,154 @@ def active_hops(nhg):
     else:
-        for nhg_prefix, alias in fg_nhg_alias.items():
-            if nhg == alias:
-                if ":" in nhg_prefix:
-                    for key in table_keys:
-                        mod_key = key.split("|")[1].split("/")[0]
-                        mod_nhg_prefix = nhg_prefix.split("/")[0]
-                        if ipaddress.ip_address(mod_key).exploded == ipaddress.ip_address(mod_nhg_prefix).exploded:
-                            t_dict = state_db.get_all(state_db.STATE_DB, key)
-                    nhg_prefix = "FG_ROUTE_TABLE|" + nhg_prefix
-                else:
-                    nhg_prefix = "FG_ROUTE_TABLE|" + nhg_prefix
-                    t_dict = state_db.get_all(state_db.STATE_DB, nhg_prefix)
-
+        nhip_prefix_map = {}
+        header = ["FG NHG Prefix", "Active Next Hops"]
+        try:
+            fg_nhg_member_table = config_db.get_table('FG_NHG_MEMBER')
+        except Exception as e: 
+            ctx.fail("FG_NHG_MEMBER entries not present in config_db")
+        alias_list = []
+        nexthop_alias = {}
+        output_list = []
+        for nexthop, nexthop_metadata in fg_nhg_member_table.items():
+            alias_list.append(nexthop_metadata['FG_NHG'])
+            nexthop_alias[nexthop] = nexthop_metadata['FG_NHG']
+        if nhg not in alias_list:
+            ctx.fail("Please provide a valid NHG alias")
+        else:
+            for nhg_prefix in table_keys:
+                t_dict = state_db.get_all(state_db.STATE_DB, nhg_prefix)
                 vals = sorted(set([val for val in t_dict.values()]))
-
                 for nh_ip in vals:
-                    if nhg_prefix in output_dict:
-                        output_dict[nhg_prefix].append(nh_ip.split("@")[0])
+                    nhip_prefix_map[nh_ip.split("@")[0]] = nhg_prefix
+                   
+                    if nh_ip.split("@")[0] in nexthop_alias:
+                        if nexthop_alias[nh_ip.split("@")[0]] == nhg:
+                            output_list.append(nh_ip.split("@")[0])
                     else:
-                        output_dict[nhg_prefix] = [nh_ip.split("@")[0]]
-
-                nhg_prefix_report = (nhg_prefix.split("|")[1])
-                formatted_nhps = ','.replace(',', '\n').join(output_dict[nhg_prefix])
-                table.append([nhg_prefix_report, formatted_nhps])
-                click.echo(tabulate(table, header, tablefmt="grid"))
-
+                        ctx.fail("state_db and config_db have FGNHG prefix config mismatch. Check device config!");
+                output_list = sorted(output_list)
+            if not output_list:
+                ctx.fail("FG_ROUTE table likely does not contain the required entries")
+            nhg_prefix_report = nhip_prefix_map[output_list[0]].split("|")[1]
+            formatted_output_list = ','.replace(',', '\n').join(output_list)
+            table.append([nhg_prefix_report, formatted_output_list])
```

Comment:
Is the output format updated in the HLD? https://github.com/Azure/SONiC/blob/master/doc/ecmp/fine_grained_next_hop_hld.md#232-cli-show-fgnhg-active-hops-fg-nhg-group-name-output-format

Reply:
Yes there is a PR for that also. https://github.com/Azure/SONiC/pull/854